### DISCUSSION TITLE
fileserver: Connection reset by peer

### DISCUSSION THREAD

**Discussion by jayl1e** [PingCAP-QE/ci#2557] [Ideas]
during curl both get and post: `curl: (56) Recv failure: Connection reset by peer`

**Comment by wuhuizuo** [comment]
@purelind and @wuhuizuo talked about this offline, and we suggest refactor uploading and downloading steps:
use ks3util tool rather than "curl" or "wget" tools.

**Reply by purelind** [reply]
Example to upload or downlod file with ks3util: 
* upload single file
```shell
# 例1: 上传后的对象为ks3://samplebucket/abc/sample.txt
./ks3util cp /home/sample.txt ks3://samplebucket/abc/

# 例2: 上传后的对象为ks3://samplebucket/abc
./ks3util cp /home/sample.txt ks3://samplebucket/abc
```
* download single file
```shell
./ks3util cp ks3://samplebucket/src-prefix/sample.txt local-dir/sample-file.txt
```

**Reply by purelind** [reply]
For more info please refer https://docs.ksyun.com/documents/42067

**Comment by purelind** [comment]
build pipeline need update:
1. build-common
2. docker-common
